### ISSUE TITLE
About rotation loss dynamics

### ISSUE THREAD

**Issue by amorehead** [jasonkyuyim/se3_diffusion#18]
Hello.

I was curious if you have observed any particular behaviors that you think are worth noting when it comes to our rotation and translation diffusion losses. Specifically, when you train your own models, do you notice the translation loss being easier for the model to optimize, or are both loss terms equally challenging for the model?

**Comment by jasonkyuyim** [comment]
Hi, I believe the translation is harder to optimize. Once the model gets the translations right, the rotations becomes easy. I've done mini experiments of fixing rotations or translations then trying to learn the other component. Rotations were learned immediately. I recommend using the wandb evaluation set-up in the code base. The losses don't tell you much. The best way to see how well you're doing is to run sampling between training epochs and evaluate metrics like steric clashes and secondary structure composition.

**Comment by amorehead** [comment]
Hi, @jasonkyuyim. I hope you are doing well. I was curious if you have also observed (what appear to be) instabilities in the training `rot_loss` for this model over the course of the two weeks it takes to train a model to convergence. The picture below shows my loss curves for this model after two days of training on 2 A100 GPUs using DDP. Interestingly, below this picture are all my other training losses, which all appear to be steadily decreasing.
![image](https://github.com/jasonkyuyim/se3_diffusion/assets/7051982/10955276-2ee6-42e3-a9e7-9f4e3ec69b74)
![image](https://github.com/jasonkyuyim/se3_diffusion/assets/7051982/479ac2e9-d9ee-42bd-911c-01224533587d)
![image](https://github.com/jasonkyuyim/se3_diffusion/assets/7051982/80c73f19-8109-42d0-ab5e-954502f6117d)
![image](https://github.com/jasonkyuyim/se3_diffusion/assets/7051982/9b71295b-865c-40b4-9fb5-45282ef1bc60)

In your experience, is this expected behavior? Are the rotation loss dynamics simply much harder to optimize compared to translations or backbone atom positions?

**Comment by jasonkyuyim** [comment]
Hi, we just posted a update that affects the rotation score learning. Please take a look at our README. I'm not sure if it's related to your problem but it might help. Though I must say I've never seen the rotation loss get that high. Are you sure you didn't change anything from the repo?

**Comment by amorehead** [comment]
Thanks, @jasonkyuyim. I will take a closer look at the updates shortly. I appreciate your quick response here. I'll close this issue for now, as I suspect it will be addressed with the rotation loss updates. Great catch, BTW!

**Comment by jamaliki** [comment]
Hi @amorehead ,

Did you manage to get everything working with Lightning and your sampler? Is it appreciably faster, because if it is do you mind sharing the code?

Best,
Kiarash.

**Comment by jasonkyuyim** [comment]
Hi, the frameflow code is probably of interest to you. I started from FrameDiff and refactored it to work with pytorch lightning and support for DDP (training and inference). A future update will add SE(3) diffusion to frameflow.
https://github.com/microsoft/flow-matching

**Comment by jamaliki** [comment]
Thanks @jasonkyuyim , I saw this and am taking a look. Great work!

**Comment by amorehead** [comment]
Agreed. Great work!

**Comment by a-green-hand-jack** [comment]
> Hi, the frameflow code is probably of interest to you. I started from FrameDiff and refactored it to work with pytorch lightning and support for DDP (training and inference). A future update will add SE(3) diffusion to frameflow.

Hello, I want to know if you have made this part of the code public? If so, where can I find it?
